src/main/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/SortClauseDeserializer.java

```diff
@@ -85,6 +85,21 @@ public SortClause deserialize(JsonParser parser, DeserializationContext ctxt)
           sortExpressions.add(exp);
           break;
         }
+      } else if (DocumentConstants.Fields.VECTOR_EMBEDDING_TEXT_FIELD.equals(path)) {
+        if (totalFields > 1) {
+          throw new JsonApiException(
+              ErrorCode.VECTOR_SEARCH_USAGE_ERROR,
+              ErrorCode.VECTOR_SEARCH_USAGE_ERROR.getMessage());
+        }
+        if (!inner.getValue().isTextual()) {
+          throw new JsonApiException(
+              ErrorCode.SHRED_BAD_VECTORIZE_VALUE,
+              ErrorCode.SHRED_BAD_VECTORIZE_VALUE.getMessage());
+        }
+        SortExpression exp = SortExpression.vectorizeSearch(inner.getValue().textValue());
+        sortExpressions.clear();
+        sortExpressions.add(exp);
```

Comment:
Is there a problem (both here and for vector field) that one could reverse order and avoid failure?
That is, there first sort clause if for `$vector`/`$vectorize`, but there's second one for "regular" property? It seems like that would not result in failure here.

Reply:
it checks as `totalFields > 1` where totalFields is size of incoming expressions.